wallet/src/Pos/Wallet/Web/Pending/Worker.hs

```diff
@@ -0,0 +1,174 @@
+    ( startPendingTxsResubmitter
+    ) where
+
+import           Universum
+
+import           Control.Monad.Catch          (handleAll)
+import           Data.Time.Units              (Second, convertUnit)
+import           Formatting                   (build, sformat, shown, (%))
+import           Mockable                     (delay, fork)
+import           Serokell.Util.Text           (listJson)
+import           System.Wlog                  (logError, logInfo, modifyLoggerName)
+
+import           Pos.Client.Txp.Addresses     (MonadAddresses)
+import           Pos.Communication            (submitAndSave)
+import           Pos.Communication.Protocol   (SendActions (..))
+import           Pos.Core                     (FlatSlotId, SlotId (..), getSlotCount)
+import           Pos.Core.Slotting            (flattenSlotId)
+import           Pos.Crypto                   (WithHash (..))
+import           Pos.Slotting                 (getLastKnownSlotDuration, onNewSlot)
+import           Pos.Txp                      (ToilVerFailure (..), TxAux (..),
+                                               getMemPool, processTx, runDBToil,
+                                               runToilTLocal, topsortTxs)
+import qualified Pos.Wallet.Web.Mode
+import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition (..))
+import           Pos.Wallet.Web.State         (casPtxCondition, getPendingTxs)
+
+type MonadPendings m =
+    ( m ~ Pos.Wallet.Web.Mode.WalletWebMode
+    , MonadAddresses m  -- TODO [CSM-407]: for now there is no way to know
+                        -- about this instance here
+    )
+
+-- | What should happen with pending transaction if attempt
+-- to resubmit it failed.
+processPtxFailure :: MonadPendings m => PendingTx -> ToilVerFailure -> m ()
+processPtxFailure PendingTx{..} e =
+    -- If number of 'ToilVerFailure' constructors will ever change, compiler
+    -- will complain - for this purpose we consider all cases explicitly here.
+    case e of
+        ToilKnown                -> tryLater
+        ToilTipsMismatch{}       -> tryLater
+        ToilSlotUnknown          -> tryLater
+        ToilOverwhelmed{}        -> tryLater
+        ToilNotUnspent{}         -> discard
+        ToilOutGTIn{}            -> discard
+        ToilInconsistentTxAux{}  -> discard
+        ToilInvalidOutputs{}     -> discard
+        ToilInvalidInputs{}      -> discard
+        ToilTooLargeTx{}         -> discard
+        ToilInvalidMinFee{}      -> discard
+        ToilInsufficientFee{}    -> discard
+        ToilUnknownAttributes{}  -> discard
+        ToilBootDifferentStake{} -> discard
+  where
+    tryLater = pass
+    discard  = do
+        casPtxCondition ptxTxId PtxApplying (PtxWon'tApply $ sformat build e)
+        logInfo $ sformat ("Transaction "%build%" was canceled") ptxTxId
+
+filterApplicablePtxs
+    :: MonadPendings m
+    => SlotId -> [PendingTx] -> m [PendingTx]
+filterApplicablePtxs curSlot ptxs = do
```

Comment:
TL;DR: simply get rid of this function.

First, this code seems simply incorrect, because you provide `mempty` as `UtxoModifier` to `runToilTLocal` down there, which means that `processTx` won't consider valid txOuts from mempool. Use `txProcessTransaction` instead.

Second - I don't believe that we really need to use block semaphore here. It seems like in theory such a conflict can appear, but it applies to basic transaction sending too, and we don't use block semaphore when we send a transaction, and we're doing it just fine so far.

Third, I actually think that you might not need to apply transactions to mempool before resubmitting it at all, because `submitAndSave` applies them anyway, and I don't believe that resending non-applicable transaction to your neighbors _once_ (because after failed resubmittion it would be deleted from the pool) is too much overhead. Moreover, you'll receive `ToilKnown` every time you try to `submitAndSave` a transaction which is already in mempool, and then (for some reason) try to resubmit it later (judging by the code above). It doesn't make sense to me.